frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -246,7 +295,7 @@ function ReaderBookmark:fixBookmarkSort(config)
     -- are not well sorted. We need to do a whole sorting for at least once.
     if config:hasNot("bookmarks_sorted") then
         table.sort(self.bookmarks, function(a, b)
-            return self:isBookmarkInPageOrder(a, b)
+            return self:isBookmarkInPositionOrder(a, b)
```

Comment:
Don't you think you need to have this re-done once for past books, so you're sure all bookmarks are precisely sorted?
If not, and if we have past highlights in a same page in that order (let's say these are `y`) [10, 30, 20, 40], I'm not sure where your addBookmark may insert a new highlight at y=25: before the 30 or after the 20 ? and the order will stay being not ordered.
`if config:hasNot("bookmarks_sorted_20220106") then` ?

Reply:
I propose to add a button `Sort bookmarks` to the extra button menu, to allow a user to sort if he needs (maybe he doesn't want).
And to remove `bookmarks_sorted` from doc_settings.

Reply:
> I propose to add a button Sort bookmarks to the extra button menu, to allow a user to sort if he needs (maybe he doesn't want).

Mhhh, no, this doesn't feel it should be a user choice. It's a technical internal ordering of the data structure, for efficiency purpose, to have an accurate binary search.
If "he doesn't want", I guess he wants to see bookmarks ordered by date added - but then it would still be by page, and only bookmarks inside a same page would be by "date added". So, a user facing button should be: `Sort by page (and position inside a same page)` / `Sort by date added` (with some new work for you) - but I think sort by date added is a bit useless :)

> And to remove bookmarks_sorted from doc_settings.

And to what aim ?:
- still calling fixBookmarkSort and have table.sort(...isBookmarkInPositionOrder) done _on each book opening_ ? (feels like an expensive no-op after you have done it once, and you have addBoookmark always insert correctly.)
- or not doing any sorting at all ?

Reply:
> * or not doing any sorting fix at all

I thought of that, and a button for manual sorting. OK, dropped.

Reply:
> config:hasNot("bookmarks_sorted_20220106")

Yes, I like the idea, re-sort all.